build.gradle

```diff
@@ -379,6 +379,7 @@ project(':iceberg-aws') {
 
     testImplementation 'software.amazon.awssdk:iam'
     testImplementation 'software.amazon.awssdk:s3control'
+    testImplementation 'com.twitter:chill-java'
```

Comment:
Why is this needed?

Reply:
This is needed because kryo serializer by default is not self-sufficient to de-serialize unmodifiable map, serializers for un-modifiable map comes from twitter's chill library when spark tries to ser/de. Hence to mimic the same behavior without  making aws module taking a dependency on spark, I added this dependency. (more details -  [here](https://spark.apache.org/docs/latest/tuning.html#data-serialization))

Reply:
That means that this is not actually serializable with Kryo, then.

Looks like the problem is that you're using `SerializableMap.copyOf(properties).immutableMap()`. Instead, you should store the copy and call `immutableMap()` in the `properties()` method to return the map to the caller.